docs/maintenance.md

```diff
@@ -0,0 +1,99 @@
+  - Read the [release note](https://github.com/container-storage-interface/spec/blob/master/spec.md) and check all the changes from the current version to the latest.
+  - Basically, CSI spec should NOT be upgraded aggressively in this task.
+    Upgrade the CSI version only if new features we should cover are introduced in newer versions, or the Kubernetes versions TopoLVM is going to support does not support the current CSI version.
+4. CSI sidecars
+  - TopoLVM does not use all the sidecars listed [here](https://kubernetes-csi.github.io/docs/sidecar-containers.html).
+    Have a look at `csi-sidecars.mk` first and understand what sidecars are actually being used.
+  - Check the release pages of the sidecars under [kubernetes-csi](https://github.com/kubernetes-csi) one by one and choose the latest version for each sidecar which satisfies both "Minimal Kubernetes version" and "Supported CSI spec versions".
+    DO NOT follow the version compatibility tables in this [page](https://kubernetes-csi.github.io/docs/sidecar-containers.html) and the README.md files in the sidecar repositories because they are sometimes not updated properly.
+  - Read the change logs which are linked from the release pages.
+
+Please write down to the Github issue of this task what kinds of changes we find in the release note and what we are going to do and NOT going to do to address the changes.
+The format is up to you, but this is very important to keep track of what changes are made in this task, so please do not forget to do it.
+
+Basically, we should pay attention to breaking changes and security fixes first.
+If we find some interesting features added in new versions, please consider if we are going to use them or not and make a GitHub issue to incorporate them after the upgrading task is done.
+
+### Update written versions
+
+Once we decide the versions we are going to upgrade, we should update the versions written in the following files manually.
+
+- `README.md`: Documentation which indicates what versions are supported by TopoLVM
+- `deploy/README.md`: Documentation which instructs how to deploy TopoLVM on various versions of Kubernetes
+- `csi-sidecars.mk`: Makefile for building CSI sidecars
+- `Makefile`: Makefile for running e2e tests
+- `e2e/Makefile`: Makefile for running e2e tests
+- `example/Makefile`: Makefile for running example
+- `deploy/manifests/overlays/daemonset-scheduler/kustomization.yaml`: Kustomization for overwriting the TopoLVM image version
+- `deploy/manifests/overlays/deployment-scheduler/kustomization.yaml`: Kustomization for overwriting the TopoLVM image version
+
+`git grep 1.18`, `git grep image:`, and `git grep VERSION` might help us avoid overlooking necessary changes.
+Please update the versions in the code and docs with great care.
+
+### Update CSI spec
+
+In case we update the version of the CSI spec written in `Makefile`, we should regenerate the gRPC schema with the protobuf file corresponding to the CSI spec version.
```

Comment:
To update **CSI** version, we need to update `CSI_VERSION` in Makefile, and run the following commands:

```console
$ make csi.proto
$ make generate 
```